Form/FormFlow.php

```diff
@@ -712,7 +758,13 @@ protected function applyDataFromSavedSteps() {
 	 * {@inheritDoc}
 	 */
 	public function createForm() {
-		return $this->createFormForStep($this->currentStepNumber);
+		$form = $this->createFormForStep($this->currentStepNumber);
+
+		if ($this->isStale()) {
+			$form->addError(new FormError('This form has expired. Please resubmit your information.'));
```

Comment:
The error message must be made translatable. Could you try to use the same approach as `PreviousStepInvalidEvent` and `PreviousStepInvalidEventListener`?